infra/terraform/modules/data_access/memberships/memberships.py

```diff
@@ -52,10 +53,35 @@ def attach_bucket_policy(event, context):
     team_slug = event["team"]["slug"]
 
     client = boto3.client("iam")
-    client.attach_role_policy(
-        RoleName=naming.role_name(username),
-        PolicyArn=policy_arn(team_slug, policy_type),
-    )
+
+    def fn():
+        client.attach_role_policy(
+            RoleName=naming.role_name(username),
+            PolicyArn=policy_arn(team_slug, policy_type),
+        )
+
+    retry(fn)
+
+
+def retry(fn, attempts=5, delay=0.150):
+    tries = attempts
+    while tries > 0:
```

Comment:
This test is applied twice. It doesn't need to be here. This can be `while True` (unless you want to call `retry` with `attempts=0`, which makes no sense).